### ISSUE TITLE
Corrupted image created during auto upload

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1350]
Received via email:
`I have been having an issue with my photos using this app. Whenever I take a photo the next cloud app creates a grey burst mode photo that is corrupt. I have to delete this photo everytime as it won't backup or open.
`

`The corrupted photo is created on my Android device every time I take a photo. I have a Nexus 6p running Android 7.1.2 with August 2017 patches.  I am running the beta version of the next cloud app to test a battery issue that I reported. This issue was also happening before that and with the help of Google we were able to narrow the issue down to this app. 
`

I am currently clueless. The autoupload should only, upload a picture found in the source folder and does not create a new one.
@mario, maybe you have an idea?

![screenshot_20170814-071350-01](https://user-images.githubusercontent.com/5836855/29348991-9395d430-8257-11e7-9a32-3d55e3addd5f.jpeg)
![screenshot_20170814-071400](https://user-images.githubusercontent.com/5836855/29348992-93985f34-8257-11e7-9611-f766f119ae9f.png)
![screenshot_20170814-071404](https://user-images.githubusercontent.com/5836855/29348993-9399c716-8257-11e7-9486-11d7b0679f99.png)

**Comment by tobiasKaminsky** [comment]
Additional info:

> If I take two pictures they both upload fine, and a third corrupt photo is left on my phone which does not upload.

**Comment by mario** [comment]
Are we talking about images stored to an SD card or internal storage? Because if on SD card, the image is copied to our own directory, locked and then uploaded. Afterwards, the temporary copy is deleted. As for internal storage, we don't do this dance as I can lock the file there.

However I've never had a case of corrupted image. So we need to find a way to reproduce this...

**Comment by ralieghnet** [comment]
I'm also having the same problem. Here is a screen shot right after the upload happens. I'm using a Samsung S6 edge. 
App version 1.4.3
I've done some testing and the error will occur regardless of internet connectivity or setting based on connectivity. It occurs each time I take a photo or place a picture in the selected directory 
local
/storage/emulated/0/DCIM/
remote
/InstandUpload/Camera

![screenshot_20171002-145324](https://user-images.githubusercontent.com/32470428/31103489-a34a22c2-a783-11e7-8771-a3e9cff5700a.png)

**Comment by AndyScherzinger** [comment]
@ralieghnet please give 2.0.0RC8 a try since 1.4.3 is obsolete with the upcoming 2.0.0

**Comment by loxK** [comment]
Have the issue following my S7 upgrade to android 8.0.0 running latest Nextcloud app from playstore. Pre android upgrade I face the issue that only a few photos and no videos were ever uploaded #2019

**Comment by meredrica** [comment]
For me, auto upload actually destroyed an image. That's a big issue. I wonder why this can happen but the local image is actually broken now.

Remote copy is of course the same.
Its an image in the Instagram folder, attached below. The original image was not set to backup to Google so basically I lost it. 

Not a big deal for this one, but this male's me very nervous.

Syncing via Web dav (using folder sync pro) is not an option because large files(videos over 4gb) always time out and are re uploaded. (I burned my data plan with that, 16gb gone within two hours. Stupid me, should have enabled wifi only)

BTW, this image is not anonymized. It's literally what nextcloud did to it. 

![IMG_20190124_230933_472](https://user-images.githubusercontent.com/450950/55667401-17244800-585c-11e9-86f3-5cf17ddb41b6.jpg)

**Comment by meredrica** [comment]
well. BUMP

**Comment by Obi-Wahn** [comment]
I get this error since yesterday. I have a pixel 3 with Android 10 and Nextcloud App Version 3.12.0 and Nextcloud server 18.0.6.
I take a picture and after the Auto Upload is done, a grey and corrupt image is showing up. The first Screenshot is from the location of the original image, the second one from the grey image. The third image is from the gallery.

![Screenshot_20200614-083858](https://user-images.githubusercontent.com/783588/84587014-f3736f00-ae1b-11ea-8f92-374f531d2465.png)
![Screenshot_20200614-083922](https://user-images.githubusercontent.com/783588/84587016-f4a49c00-ae1b-11ea-8a61-4b8f0f022418.png)
![Screenshot_20200614-084038](https://user-images.githubusercontent.com/783588/84587018-f53d3280-ae1b-11ea-9cd2-0a6b5eb66ff5.png)

**Comment by AndyScherzinger** [comment]
@Obi-Wahn your issue seems unrelated to Nextcloud since we don't create any of the mentioned files/images so the second, grey image must be created by some other app or the OS and be a bug some place else, Sorry 😢

**Comment by Obi-Wahn** [comment]
@AndyScherzinger Thanks for the answer.
But it seems to me that there is a connection after all. only when the synchronization of Nextcloud is active, the grey pictures appear. When the sync is off, no gray images appear when I take pictures. I don't know what other app it could be. i have just tried reinstalling the app, but it did not work.

**Comment by AndyScherzinger** [comment]
Strange, the only thing I can think of would be some file transfer between the 2 folders by the camera app itself while the Nextcloud app would already access the file interfering with the camera app's file system operations.
What do you think @tobiasKaminsky @ezaquarii

**Comment by hammy1988** [comment]
I have the same problem on my Huawei P30 Pro (VOG-L29, Android 10).

I tested it with my old Samsung S8 (SM-G950F, Android 9) and Huawei Y7 (DUB-LX1, Android 8.1).
On this phones is can't reproduce the issue.

I borrowed a second Huawei P30 Pro and on this phone is the issue, too.

I created a local Folder, copied some photos in there and uploaded it to the next cloud.
On the Huawei P30 Pro (with Android 10) is the error which @AndyScherzinger described.

Huawei Photo-App listed the "dead" photos, but i can't find them on the phone storage.
I try to upload the "internal storage"/Pictures, so Nextcloud tried to uploaded the duplicated, too.

![Screenshot_20200828_193653_com nextcloud client](https://user-images.githubusercontent.com/30080459/91601041-4499f600-e969-11ea-86d3-db97bb3d9974.jpg)
Translated: local data not found

![Screenshot_20200828_193717_com android gallery3d](https://user-images.githubusercontent.com/30080459/91601064-4e235e00-e969-11ea-9b09-4dabb0f07c48.jpg)
The date of the "dead" photo is the date the original photo was uploaded by Nextcloud.

It can't be a other app, because i stopped the Nextcloud-App and triggert the upload some minutes late on another test and the createn time (time of the photo) in the Huawei Gallery is the exact time when nextcloud uploaded the photo.

**Comment by hammy1988** [comment]
I completely reset my second smartphone (Huawei P30 Pro, Android 10). 
So no other app could interfere.

But the same error occurs again with every test image that is uploaded by Nextcloud.

![Zwischenablage02](https://user-images.githubusercontent.com/30080459/91658946-b9d20c00-eacc-11ea-8d24-906527e0dee9.jpg)

![Zwischenablage01](https://user-images.githubusercontent.com/30080459/91658949-bc346600-eacc-11ea-8883-0049ec03eb0c.jpg)

**Comment by tobiasKaminsky** [comment]
Thanks for the detailed debugging, @hammy1988 
What option do you have for auto upload? copy/move/delete?
Can you set it to copy, so that at least original images should not vanish.

Do you have any battery saving mechanism in P30?

**Comment by hammy1988** [comment]
NextCloud has no battery restrictions. It is excluded from the Huawei battery management.

I only testet with copy (keep original).
I did some additional tests.

The error occures with this configuration
- [x] copy (keep original)
- [ ] move (move to app-folder)
- [x] delete

Only with the configuration "move" the NextCloud-App do not create a duplicate corrupted image in the folder ```Internal Storage/Pictures```.
I tested it several times with different folders and photos.

**Comment by hammy1988** [comment]
I don't know if it will help you. But i instelled the Nextcloud Dev App (v20200829) and get a logfile.
But i can't find any copy entry that could explain the additional file.

[Log_File_Nextcloud_Dev_Version20200829.txt](https://github.com/nextcloud/android/files/5151737/Log_File_Nextcloud_Dev_Version20200829.txt)

Is it possible to get deeper log files?

**Comment by tobiasKaminsky** [comment]
Thanks for the detailed debug!
I also do not see any suspicious…

From your three tests it seems that the copy/delete process is interrupted in between and thus a corrupted file is created/remains.

Without access to such a device, I fear I cannot do anything :-(

**Comment by hammy1988** [comment]
@tobiasKaminsky : i sent you an email. Did you get it? Maybe it will help to fix this bug (if you are interesseted to debug this issue).

**Comment by Obi-Wahn** [comment]
I just discovered, that the bug is gone for me. I don't know what the solution was. There are several possibilities. I upgraded my Pixel 3 to Android 11and i upgraded Nextcloud to 19.0.3. One big change could be the naming of the pictures. It changed from IMG_2020- to PXL_2020-. 
Sorry, I can't narrow it down.

**Comment by tobiasKaminsky** [comment]
Now looking into it.
@hammy1988 thanks for sending your spare device :-)

**Comment by tobiasKaminsky** [comment]
Found it.
It is a bug from NC, but only visible on Huawei devices it seems:
- we trigger media scanner all the time, even when only uploading images, thus we trigger it with "" path (as there is no new image)
- media scanner seems to ignore this on other devices
- but on huawei/gallery app it shows it with default location "/Pictures" and file name (as we pass this).

This leads then to the "corrupted" image, where in reality there is just no image at all…

@hammy1988 I now can also trigger another bug (#2010) with your device. I hope it is ok to keep it some more day? :crossed_fingers:

**Comment by hammy1988** [comment]
Hey @tobiasKaminsky,

many thanks for your work! It's nice that you also take care of errors that only occur with a small member of users.

Of course you can keep my Huawei some more days. I don't need it right now 😄

**Comment by Obi-Wahn** [comment]
@tobiasKaminsky Thanks! I hope this fixes the bug for me too.

**Comment by tobiasKaminsky** [comment]
@Obi-Wahn you can test it with APK in #6984, which you can install in parallel.
If it does not help, please open up a new issue.

**Comment by Obi-Wahn** [comment]
I'm using the Nextcloud App Version 3.13.1 and the bug seems to be gone. Thanks!
